src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java

```diff
@@ -147,6 +148,65 @@ private String getMessageWithArgs(String resourceBundle, final Locale locale, fi
         return FormatUtils.formatMsg(message, Locale.forLanguageTag(msgsItemDTO.getLocale()), args);
     }
 
+    /**
+     * Get one version's key based message if there is multiple versions for one product, the message's arguments are put in a variable object
+     *
+     * @param locale
+     * @param version
+     * @param component
+     * @param key
+     * @param args
+     * @return
+     */
+    public String getMultiVersionMessage(final Locale locale, final String version, final String component, final String key, final Object... args) {
+        return getMultiVersionMessageWithArgs(null, locale, version, component, key, args);
+    }
+
+    /**
+     * Get one version's key based message if there is multiple versions for one product, the message's arguments are put in a Map
+     *
+     * @param locale
+     * @param version
+     * @param component
+     * @param key
+     * @param args
+     * @return
+     */
+    public String getMultiVersionMessage(final Locale locale, final String version, final String component, final String key, final Map<String, Object> args) {
+        return getMultiVersionMessageWithArgs(null, locale, version, component, key, args);
+    }
+
+    private String getMultiVersionMessageWithArgs(String resourceBundle, final Locale locale, final String version, final String component, final String key, final Object args) {
+        // Use source message if the message hasn't been collected/translated
+        String source = null;
+        try {
+            source = resourceBundle != null ? ResourceBundle.getBundle(resourceBundle).getString(key) :
+                    getMultiVersionMessagesOfKey(Locale.forLanguageTag(ConstantsKeys.SOURCE), version, component, key,false).getMessages().get(version);
+        }catch(Exception e){
+            logger.error(e.getMessage());
+        }
+        if (source!=null && !source.isEmpty()) {
+            String collectedSource = getMultiVersionMessagesOfKey(LocaleUtility.getSourceLocale(), version, component, key,false).getMessages().get(version);
+            if (!source.equals(collectedSource)) {
+                return FormatUtils.formatMsg(source, LocaleUtility.getSourceLocale(), args);
+            }
+        }
+        if(locale == null){
+            throw new VIPJavaClientException(ConstantsMsg.LOCALE_CANNOT_NULL);
+        }
+        if(StringUtil.isEmpty(component)){
+            throw new VIPJavaClientException(ConstantsMsg.COMPONENT_CANNOT_EMPTY);
+        }
```

Comment:
the argument checking should be put in getMultiVersionMessage.

Reply:
Since there are two 'getMultiVersionMessage' method, which are for support different argument types, if put the argument checking code to these two methods, there will be duplicate codes, so extract the common code here.